lib/simp/build/iso/tree_info_reader.rb

```diff
@@ -0,0 +1,87 @@
+require 'puppet'
+require 'puppet/util/inifile'
+
+module Simp; end
+module Simp::Build; end
+module Simp::Build::Iso
+  # Parses data from .treeinfo files. Supports pre-productmd and treeinfo 1.x.
+  class TreeInfoReader
+    PRE_PRODUCTMD_TREEINFO_VERSION = '0.0-pre-productmd'
+
+    def initialize(treeinfo_file, target_arch: 'x86_64')
+      File.exist?( treeinfo_file ) or fail("File does not exist: '#{treeinfo_file}'")
+
+      @target_arch = target_arch
+      @file = treeinfo_file
+      @ini = Puppet::Util::IniConfig::PhysicalFile.new(@file)
+      @ini.read
+      @treeinfo_version = treeinfo_version
+      @treeinfo_maj_version = treeinfo_version.split('.').first.to_i
+    end
+
+    # release version, for example: "21", "7.0", "2.1"
+    # @return [String]
+    def release_version
+      @treeinfo_maj_version > 0 ? section('release')['version'] : section('general')['version']
```

Comment:
What if section() returns nil?  This code does not handle that error case in a fashion that an end-user would understand?